src/Control/Distributed/Process/Management/Internal/Trace/Primitives.hs

```diff
@@ -168,6 +169,10 @@ withRegisteredTracer act = do
   withLocalTracer $ \t -> liftIO $ Tracer.getCurrentTraceClient t sp
   currentTracer <- receiveChan rp
   case currentTracer of
-    Nothing  -> do { (Just p') <- whereis "tracer.initial"; act p' }
+    Nothing  -> do mTP <- whereis "tracer.initial"
+                   -- NB: this should NOT ever happen, but forcing pattern matches
+                   --     is not considered cool in later versions of MonadFail
+                   case mTP of
+                     Just p' -> act p'
+                     Nothing -> die "Initial Tracer Name Not Registered"
```

Comment:
can we add something like: `This should never happen` or `invariant violation` to the message. It would be nice if a user will get enough info to understand what had happened in this case.